### ISSUE TITLE
The `env` was cleaned in `rustc_compile_action`, which doesn't work well with tools using script beginning with `#!/usr/bin/env *`

### ISSUE THREAD

**Issue by moshimo2019** [bazelbuild/rules_rust#2242]
bazel 5.3.0
rules_rust 0.26.0
detailed log
```
ERROR: /home/username/.cache/bazel/_bazel_username/fd39ace65beb845156ae48456934004c/external/rules_rust/util/process_wrapper/BUILD.bazel:6:36: Compiling Rust (without process_wrapper) bin process_wrapper (6 files) failed: (Exit 1): linux-sandbox failed: error executing command 
  (cd /home/username/.cache/bazel/_bazel_username/fd39ace65beb845156ae48456934004c/sandbox/linux-sandbox/186/execroot/asr-decoder-capi && \
  exec env - \
    CARGO_CFG_TARGET_ARCH=x86_64 \
    CARGO_CFG_TARGET_OS=linux \
    CARGO_CRATE_NAME=process_wrapper \
    CARGO_MANIFEST_DIR='${pwd}/external/rules_rust/util/process_wrapper' \
    CARGO_PKG_AUTHORS='' \
    CARGO_PKG_DESCRIPTION='' \
    CARGO_PKG_HOMEPAGE='' \
    CARGO_PKG_NAME=process_wrapper \
    CARGO_PKG_VERSION=0.0.0 \
    CARGO_PKG_VERSION_MAJOR=0 \
    CARGO_PKG_VERSION_MINOR=0 \
    CARGO_PKG_VERSION_PATCH=0 \
    CARGO_PKG_VERSION_PRE='' \
    SYSROOT=bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/rust_toolchain \
    TMPDIR=/tmp \
  /home/username/.cache/bazel/_bazel_username/install/1a80ecdfe78ff11e1566d5084a48c3f7/linux-sandbox -t 15 -w /home/username/.cache/bazel/_bazel_username/fd39ace65beb845156ae48456934004c/sandbox/linux-sandbox/186/execroot/asr-decoder-capi -w /tmp -w /dev/shm -D -- bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/rust_toolchain/bin/rustc external/rules_rust/util/process_wrapper/main.rs '--crate-name=process_wrapper' '--crate-type=bin' '--error-format=human' '--out-dir=bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rules_rust/util/process_wrapper' '--codegen=opt-level=3' '--codegen=debuginfo=0' '--remap-path-prefix=${pwd}=' '--emit=dep-info,link' '--color=always' '--target=x86_64-unknown-linux-gnu' -L bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rust_linux_x86_64__x86_64-unknown-linux-gnu__stable_tools/rust_toolchain/lib/rustlib/x86_64-unknown-linux-gnu/lib '--edition=2018' '--codegen=linker=external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc' --codegen 'link-args=-Wl,-no-as-needed -B/usr/bin -ldl -lpthread -lstdc++ -no-canonical-prefixes -fno-canonical-system-headers -pie -Wl,-z,relro,-z,now -Wl,--gc-sections -Wl,--build-id=md5 -Wl,--hash-style=gnu' '--extern=tinyjson=bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rules_rust_tinyjson/libtinyjson-4031717389.rlib' '-Ldependency=bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rules_rust_tinyjson')
...
...
error: linking with `external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc` failed: exit status: 127
...
= note: /usr/bin/env: 'python': No such file or directory
error: aborting due to previous error
```

The script `external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc` is a script beginning with `#!/usr/bin/env python`. If I change it to `#!/usr/bin/python`, the compilation doesn't fail, while the original one is more general.

It seem that the command `exec env -` generated by rust_binary clean and set its own environmental variables ('/usr/bin' isn't included), and then other invoked tools relying on `env` invoked doesn't works well.
Is this behavior of clean all env appropriate? It there any way to repair it without modifying the affected tools?

**Comment by granaghan** [comment]
I think this is the same issues as #2177.

Unfortunately, it won't be fully cleanly resolved until bazel 7.0 where the behavior of merging environments will change. For now you could try working around it with:
`--host_action_env=PATH="$PATH" --action_env=PATH="$PATH"`

**Comment by moshimo2019** [comment]
> I think this is the same issues as #2177.
> 
> Unfortunately, it won't be fully cleanly resolved until bazel 7.0 where the behavior of merging environments will change. For now you could try working around it with: `--host_action_env=PATH="$PATH" --action_env=PATH="$PATH"`

I tried this solution but get some weird errors:
```shell
/bin/bash: dirname: command not found
/bin/bash: line 0: cd: HOME not set
```
I guess that there are some complicated terms in `$PATH`. What the compilation needs is inside `/usr/bin`, so I tried with `--host_action_env=PATH="/usr/bin" --action_env=PATH="/usr/bin"`, which works well in my case.
Thanks for your reply!!!
